shuup/core/models/_orders.py

```diff
@@ -810,8 +811,13 @@ def create_refund(self, refund_data, created_by=None, supplier=None):
             assert quantity
 
             if parent_line == "amount":
+                refund_supplier = supplier
+
+                if not refund_supplier:
+                    refund_supplier = refund.get("refund_supplier")
+
```

Comment:
What if we still do not have supplier? Is it ok?

Reply:
Yes it's okej.